contracts/contracts/BeamerUtils.sol

```diff
@@ -7,53 +7,53 @@ library BeamerUtils {
         bytes32 fillId;
     }
 
-    function createRequestHash(
-        uint256 requestId,
+    function createRequestId(
         uint256 sourceChainId,
         uint256 targetChainId,
         address targetTokenAddress,
         address targetReceiverAddress,
-        uint256 amount
+        uint256 amount,
+        uint256 nonce
```

Comment:
What was the reason we did not include `sourceTokenAddress` here originally?

Reply:
We don't need to do this. The request id is supposed to contain all information one need to verify that the request has been fulfilled. 
The source token address is nothing the user requested, it's actually something the agent requests to fulfill the service.

Reply:
Thinking further, I think in the end this may actually be fine because we have both the source chain ID _and_ the nonce there, which should suffice on its own to identify (and only identify) a request.

Reply:
due to the nonce it is strictly impossible to get the same request id twice. The main purpose of the request id is to forbid the agent to not fulfill the service correctly (i.e. sending fewer tokens, sending the wrong token) but generating a proof for the request id. 
The source token here is not of interest, the agent wouldn't simply fill it, if he isnt happy with it